dandi/tests/test_utils.py

```diff
@@ -459,3 +460,42 @@ def test_get_mime_type(filename: str, mtype: str) -> None:
 )
 def test_is_page2_url(page1: str, page2: str, r: bool) -> None:
     assert is_page2_url(page1, page2) is r
+
+
+@pytest.mark.parametrize(
+    "filename,suffixes,target",
+    [
+        ("blah.ome.zarr", {".ome", ".zarr"}, {".ome", ".zarr"}),
+        (
+            ".ome.zarr",
+            {".ome", ".zarr"},
+            {".zarr"},
+        ),  # does not take .ome as an extension in such case
+        # no hits and that is ok
+        ("blah.ome.zarr", {".ngff", ".dat"}, set()),
+        ("blah", {".ngff", ".dat"}, set()),
+        ("blah", set(), set()),
+        ("blah.ome", set(), set()),
+    ],
+)
+def test_has_suffixes(filename: str, suffixes: set, target: set) -> None:
+    # as is
+    assert has_suffixes(filename, suffixes) == target
+
+    # poor Yarik doesn't know nice way for combining multiple .parametrize
```

Comment:
Try:

```
@pytest.mark.parametrize("filename,suffixes,target", [ ... ])
@pytest.mark.parametrize("suffix_t", [list, set, tuple])
def test_has_suffixes(filename: str, suffixes: set[str], target: set[str], suffix_t: type) -> None:
```

Reply:
I didn't dare to try -- works wonderfully ;)  will push this change even though I think we might close this PR ;)